zanata-war/src/main/java/org/zanata/service/impl/CopyTransWorkFactory.java

```diff
@@ -57,13 +82,374 @@
     @In
     private VersionStateCache versionStateCacheImpl;
 
-    public CopyTransWork createCopyTransWork(HLocale targetLocale,
+    public Integer runCopyTransInNewTx(HLocale targetLocale,
+            HCopyTransOptions options, HDocument document,
+            boolean requireTranslationReview, List<HTextFlow> copyTargets)
+            throws Exception {
+        return runInTransaction(() -> runCopyTrans(targetLocale, options,
```

Comment:
And this https://wmarkito.wordpress.com/2012/07/20/cdi-events-synchronous-x-asynchronous/